Scripts/Runtime/Entities/Tasks/TaskFlow/Nodes/DataStreamNode.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    internal class DataStreamNode : AbstractNode
+    {
+        private readonly Dictionary<Type, byte> m_ResolveTargetLookup;
+        private readonly NodeLookup m_Lookup;
+
+        public AbstractEntityProxyDataStream DataStream { get; }
+        public AbstractTaskStream TaskStream { get; }
+
+        public DataStreamNode(NodeLookup lookup,
+                              AbstractEntityProxyDataStream dataStream,
+                              TaskFlowGraph taskFlowGraph,
+                              AbstractTaskSystem taskSystem,
+                              AbstractTaskDriver taskDriver,
+                              AbstractTaskStream taskStream) : base(taskFlowGraph, taskSystem, taskDriver)
+        {
+            m_Lookup = lookup;
+            DataStream = dataStream;
+            TaskStream = taskStream;
+
+            m_ResolveTargetLookup = new Dictionary<Type, byte>();
+        }
+
+        protected override void DisposeSelf()
+        {
+            DataStream.Dispose();
```

Comment:
You're right. I've update #71 to change my mind and have the ownership be in the TaskDriver/TaskSystem. 

We can't dispose the TaskStream here because there's a node for each DataStream in a TaskStream. 

I've added a TODO to fix this.